09-intro-to-bcdata.Rmd

```diff
@@ -17,9 +17,133 @@ knitr_fig_path("09-")
+
+
+```{r}
+bcdc_get_record("bc-public-libraries-systems-branches-and-locations")
+```
+
+Next, we can use the `bcdc_get_data` function to load this data directly into R:
+
+```{r}
+bcdc_get_data(record = "bc-public-libraries-systems-branches-and-locations")
+
+lib_raw <- bcdc_get_data(record = 'bc-public-libraries-systems-branches-and-locations',
+                          resource = 'ed17f111-fb39-46b3-89aa-5375592fbb01')
+```
+
+
+> ## Challenge 1 (5 minutes)
+>
+> Take a few minutes to search the B.C. Data Catalogue for data sets for a different topic. Identify and import one tabular data set into your R session > (e.g. a CSV or XLS resource).
+>
+
+
+Let's return to our libraries data. As we learned earlier in the workshop, a useful first step is to get familiar with the data.
+
+```{r}
+str(lib_raw)
+names(lib_raw)
+```
+
+Let's tidy up this data frame and find out how many books&mdash;physical materials or volumes&mdash;are available within each City:
+
+```{r}
+library(dplyr)
+
+lib_tidy <- lib_raw %>%
+  select(`Library System`, Location, City, num_books = "460x Total branch physical materials, volumes held")
+
+
+sum_books <- lib_tidy %>% 
+  group_by(City, `Library System`) %>% 
+  summarize(total_books = sum(num_books))
+```
+
+> ## Challenge 2 (10 minutes)
+>
+> Using `dplyr`, determine the total number of books within each Library System in British Columbia.
+>
+>
+>```{r}
+> sum_books %>% 
+>  group_by(`Library System`) %>% 
+>  summarize(totals = sum(total_books))
+>```
+
+
+Now that we have imported, tidied, and analyzed our library data, let's make a static data visualization to share and communicate the results with others.
+
+```{r}
+library(ggplot2)
+
+sum_books %>% 
+  ggplot(aes(total_books, City)) +
+  geom_col()
+
+sum_books %>% 
+  filter(total_books > 100000) %>% 
+  ggplot(aes(total_books, City)) +
+  geom_col()
+```
+
+> ## Challenge 3 (10 minutes)
+>
+> Iterate the above plot to include information on the number of books by Library Service. Add a title to the plot. Try other design changes to make the plot more readable.
+>
+>
+>
+>```{r}
+>library(ggplot2)
+>
+>sum_books %>% 
+>  filter(total_books > 100000) %>% 
+>  ggplot(aes(total_books, City)) +
+>  geom_col() +
+>  facet_wrap(vars(`Library System`)) +
+>  theme_minimal() +
+>  labs(x = NULL,
+>       y = NULL,
+>       title = "Total Number of Books by City & Library System in B.C.",
+>       caption = "Data sourced from the B.C. Data Catalogue")
+>
+>```
+
+
+
+```{r}
+lib_spatial <- bcdc_query_geodata("bc-public-libraries-systems-branches-and-locations") %>%
+  collect()
+
+mapview::mapview(lib_spatial)
+
+lib_spatial %>% 
+  ggplot() + 
+  geom_sf()
+
+lib_spatial %>% 
+  left_join(lib_tidy,
+            by = c("SERVICE_POINT_NAME" = "Location")) %>% 
+  ggplot() +
+  geom_sf(aes(size = num_books), colour = "blue") +
+  geom_sf(data = bcmaps::bc_bound(), fill = NA) +
```

Comment:
Using bcmaps is going to be challenging for the bookdown process here

Reply:
Maybe the mapview example is sufficient?

Reply:
Yep, I will pull it out.